proposals/declaration-expressions.md

```diff
@@ -0,0 +1,78 @@
+# Declaration expressions
+
+* [ ] Proposed
+* [ ] Prototype
+* [ ] Implementation
+* [ ] Specification: see below
+
+## Summary
+[summary]: #summary
+
+Support declaration assignments as expressions.
+
+## Motivation
+[motivation]: #motivation
+
+Allow initialization at the point of declaration in more cases, simplifying code, and allowing `var` to be used.
+```c#
+SpecialType SpecialType =>
+    (var st = type.SpecialType).IsValueType() ? SpecialType.None : st;
+```
+
+Extend `out var` declarations to allow `ref` values.
+```c#
+Convert(source, destination, ref HashSet<Diagnostic> diagnostics = null);
+```
+
+## Detailed design
+[design]: #detailed-design
+
+Expressions are extended to include declaration assignment. Precedence is the same as assignment.
+
+```antlr
+expression
+    : non_assignment_expression
+    | assignment
+    | declaration_assignment_expression // new
+    ;
+declaration_assignment_expression // new
+    : declaration_expression '=' local_variable_initializer
+    ;
+declaration_expression // C# 7.0
+    | type variable_designation
+    ;
+```
+
+The declaration assignment is of a single local.
+```c#
+F(var x, y = 2);        // error
+F(var (x, y) = (1, 2)); // error
```

Comment:
This form is likely to be supported by our relaxation of deconstruction, a separate feature.

Reply:
Removed.